wandb/sdk/wandb_settings.py

```diff
@@ -262,12 +262,26 @@ class Settings(BaseModel, validate_assignment=True):
     quiet: bool = False
     """Flag to suppress non-essential output."""
 
-    reinit: Literal["return_previous", "finish_previous"] | bool = "return_previous"
+    reinit: (
+        Literal[
+            "return_previous",
+            "finish_previous",
+            "allow",
+        ]
+        | bool
+    ) = "return_previous"
     """What to do when `wandb.init()` is called while a run is active.
 
     Options:
-    - "return_previous" (default): Return the active run.
-    - "finish_previous": Finish the active run, then return a new one.
+    - "return_previous" (default): Return the most recently created run
+        that is not yet finished. This does not update `wandb.run`; see
+        the "allow" option.
+    - "finish_previous": Finish all active runs, then return a new run.
+    - "allow": Create a new run without modifying other active runs.
+        Note that if another run already exists, `wandb.run` will continue to
+        refer to it and will not be updated to the new run. When that run
+        finishes, `wandb.run` will become `None`. This may affect some
+        integrations.
```

Comment:
i think this behavior of global run is going to cause some confusion with users, as it will require to carefully track when they create and finish each run, maybe a better solution is to disallow global run when we have multiple active runs?

Reply:
or have multiple global runs? and they need to use id or something, not sure if this approach is better, but an alternative to fully disabling global run

Reply:
hmmm i see... i guess i was thinking of integrations in third party repo, where i don't think they change these settings, but i guess there is a case for integrations in our code base.

i guess it's fine, we probably should document it and discourage certain usage patterns especially with integrations?

Reply:
We don't mention using `wandb.run` in our [guide for integrations](https://docs.wandb.ai/guides/integrations/add-wandb-to-any-library/#start-a-run), and I recently updated those docs and some other top-level docs to not use the global run.

Reply:
I updated `reinit="create_new"` to not touch `wandb.run` as discussed!